drivers/flash/Kconfig.at45

```diff
@@ -7,7 +7,7 @@ menuconfig SPI_FLASH_AT45
 	depends on DT_HAS_ATMEL_AT45_ENABLED
 	select FLASH_HAS_DRIVER_ENABLED
 	select FLASH_HAS_PAGE_LAYOUT
-	depends on SPI
+	select SPI
```

Comment:
Why has this change been made? This should not select SPI, this originally correctly depends on SPI

Reply:
CC @tejlmand

Reply:
The policy is to use `select` instead of `depends on` if a peripheral uses a bus. See [this commit](https://github.com/zephyrproject-rtos/zephyr/commit/df81fef94483da9f2811d48088affbcfd61ab18c) for more info.

Reply:
That specifically states for devices that can be on an SPI and/or I2C, of which the at45 cannot as it is SPI only

Reply:
Yes, but it has been generally applied to all peripherals because it reduces the need to have additional `kconfig` symbols defined, which can be deduced from the device tree.
As the following search shows (these commits refer to the same commit):
```
$ git log --oneline --grep "df81fef"
d36125654e drivers: ethernet: Update Kconfig
a50814dc9d drivers: w1: use select SERIAL in Kconfig for w1-zephyr-serial driver
baaf904885 video: Move to using select in Kconfig for I2C bus
b37f393079 drivers: audio: Update Kconfig
e438da8a36 adc: Move to using select in Kconfig for I2C & SPI busses
18a1db193d drivers: dac: Update Kconfig for 'select I2C/SPI'
465b7615a9 drivers: w1: Move to using select in Kconfig for I2C bus
9e9aab3bcf drivers: can: Move to using select in Kconfig for SPI bus
a96ef8a631 drivers: misc: ft8xx: Update Kconfig
d9882ff422 eeprom: Move to using select in Kconfig for I2C & SPI busses
616342d58b drivers: ieee802154: Move to using select in Kconfig for SPI bus
db0a8c63fe drivers: led: Update Kconfig
287043cddb drivers: kscan: Update Kconfig
902f42f491 drivers: display: Update Kconfig
```

Reply:
> Why has this change been made? This should not select SPI, this originally correctly depends on SPI

@galak If this is the confirmed, I will close the PR.

Reply:
I will leave it for @tejlmand to comment on this. This is the exact issue that has to be fixed elsewhere because select is the wrong thing to use, depends on is correct e.g. https://github.com/zephyrproject-rtos/zephyr/pull/53778/files

Reply:
The policy to use `select` over `depends on` for bus dependencies was discussed in https://github.com/zephyrproject-rtos/zephyr/pull/48707. The rationale is, that if a user enabled a Kconfig for a given driver, surely they also want to enable to needed bus driver class.

Reply:
@henrikbrixandersen not sure why I was not included in the discussion of #48707 ?

There are some risks associated with such a transition to `select` compared to a `depends on`.
Mainly that a downstream user cannot get rid of a `select`, but it's possible to re-define a symbol that has a `depends on` and thereby allow the setting to be available when alternative dependencies are met.

However I haven't looked into details regarding #48707 so above is based on general recommendations and experiences.

Reply:
> Mainly that a downstream user cannot get rid of a `select`, but it's possible to re-define a symbol that has a `depends on` and thereby allow the setting to be available when alternative dependencies are met.

A downstream user can get rid of it, but only if the peripheral that selects the bus is `disabled` in the device tree or the driver `kconfig` symbol is manually set to `n`.

Reply:
I had a check how this is done in linux, and it is done using `depends on`, here are some examples:
```
config TOUCHSCREEN_AD7879_I2C
        tristate "support I2C bus connection"
        depends on TOUCHSCREEN_AD7879 && I2C
        select REGMAP_I2C
        help
          Say Y here if you have AD7879-1/AD7889-1 hooked to an I2C bus.

          To compile this driver as a module, choose M here: the
          module will be called ad7879-i2c.

config TOUCHSCREEN_AD7879_SPI
        tristate "support SPI bus connection"
        depends on TOUCHSCREEN_AD7879 && SPI_MASTER
        select REGMAP_SPI
        help
          Say Y here if you have AD7879-1/AD7889-1 hooked to a SPI bus.

...

config MOUSE_PS2_SYNAPTICS_SMBUS
        bool "Synaptics PS/2 SMbus companion" if EXPERT
        default y
        depends on MOUSE_PS2 
        depends on I2C=y || I2C=MOUSE_PS2
        select MOUSE_PS2_SMBUS

...

config VIDEO_IMX219
        tristate "Sony IMX219 sensor support"
        depends on I2C && VIDEO_DEV
        select MEDIA_CONTROLLER
        select VIDEO_V4L2_SUBDEV_API
        select V4L2_FWNODE

...

menuconfig MEDIA_SUPPORT
        tristate "Multimedia support"
        depends on HAS_IOMEM

if MEDIA_SUPPORT

config MEDIA_SUBDRV_AUTOSELECT
        bool "Autoselect ancillary drivers (tuners, sensors, i2c, spi, frontends)"
        depends on HAS_IOMEM
        select I2C
        select I2C_MUX
        default y if MEDIA_SUPPORT_FILTER

...

config LTC2983
        tristate "Analog Devices Multi-Sensor Digital Temperature Measurement System"
        depends on SPI
        select REGMAP_SPI
```